[llvm-debuginfo-analyzer] Fix crash with thread local storage.

CarlosAlbertoEnciso opened the pull request:
The DW_OP_GNU_push_tls_address, DW_OP_form_tls_address DWARF
location forms generated for thread local storage variables, caused a
crash in the DWARFReader, due to incorrect number of operands.

CarlosAlbertoEnciso committed:
[llvm-debuginfo-analyzer] Fix crash with thread local storage.

The DW_OP_GNU_push_tls_address, DW_OP_form_tls_address DWARF
location forms generated for thread local storage variables,
caused a crash in the DWARFReader, due to incorrect number of
operands.

jryans commented on llvm/test/tools/llvm-debuginfo-analyzer/DWARF/Inputs/ThreadLocalStorage.s:
```diff
@@ -0,0 +1,309 @@
+	.text
```
Perhaps this test could be minimised further...? It would be nice to reduce this down to just what's needed for the issue.

It may also be worth considering the `yaml2obj` format that some debug info tests use elsewhere.

CarlosAlbertoEnciso replied:
Thanks for your comments.
In the initial patches for `llvm-debuginfo-analyzer` I used `yaml2obj`. But for the latest `WebAssembly` patches, I start using `llvm-mca`.

CarlosAlbertoEnciso replied:
@jryans Changed from using a full `.s` to a `reduced IR`, with just the minimum code to reproduce the issue. Thanks.

jryans replied:
Thanks, the IR version looks much more minimal and maintainable.

jryans reviewed (commented):
Looks good overall, just wondering a bit about the test.

CarlosAlbertoEnciso committed:
[llvm-debuginfo-analyzer] Fix crash with thread local storage.

The DW_OP_GNU_push_tls_address, DW_OP_form_tls_address DWARF
location forms generated for thread local storage variables,
caused a crash in the DWARFReader, due to incorrect number of
operands.

Change the test case from being a full .s to a reduced IR.

jryans reviewed (approved):
Looks good to me, thanks for the fix! 😄

CarlosAlbertoEnciso commented:
@jryans Many thanks for your feedback.
